tests/orion/orchestration/test_core_policy.py

```diff
@@ -1553,3 +1552,321 @@ async def test_concurrency_race_condition_limit_decreases_dont_impact_existing_r
         # the concurrency slot is released as expected
         assert task1_completed_ctx.response_status == SetStateStatus.ACCEPT
         assert (await self.count_concurrency_slots(session, "shrinking limit")) == 0
+
+    async def test_returning_concurrency_slots_when_transitioning_out_of_running_even_on_fizzle_with_release_first(
```

Comment:
super pedantic: `fizzling` implies that the slot was released before the state mutation, so we don't need the `with_release_first` piece to the test name